beacon_node/network/src/beacon_processor/worker/rpc_methods.rs

```diff
@@ -204,6 +204,63 @@ impl<T: BeaconChainTypes> Worker<T> {
         )
     }
 
+    /// Handle a `BlocksByRoot` request from the peer.
+    pub fn handle_light_client_bootstrap(
+        self,
+        executor: TaskExecutor,
+        peer_id: PeerId,
+        request_id: PeerRequestId,
+        request: LightClientBootstrapRequest,
+    ) {
+        // Fetching blocks is async because it may have to hit the execution layer for payloads.
+        executor.spawn(
+            async move {
+                let state_root = request.root;
+                let mut beacon_state = match self.chain.get_state(&state_root, None) {
+                    Ok(beacon_state) => match beacon_state {
+                        Some(state) => state,
+                        None => {
+                            self.send_error_response(
+                                peer_id,
+                                RPCResponseErrorCode::ResourceUnavailable,
+                                "Bootstrap not avaiable".into(),
+                                request_id,
+                            );
+                            return;
+                        }
+                    },
+                    Err(_) => {
+                        self.send_error_response(
+                            peer_id,
+                            RPCResponseErrorCode::ResourceUnavailable,
+                            "Bootstrap not avaiable".into(),
+                            request_id,
+                        );
+                        return;
+                    }
+                };
```

Comment:
The `root` in the request is a block root rather than a state root, so we need to look up the block and then the state. A blinded block without an execution payload is sufficient, which we can fetch using `chain.get_blinded_block`.

To get the state in an efficient way we should use the snapshot cache if possible. I've added a method for doing this to this PR that I hope to merge before we add RPC: https://github.com/sigp/lighthouse/pull/3714 (you can rebase on it preemptively if you like).

The other thing in that branch is a config option to enable and disable the light client server. In case it's disabled (check `self.chain.config.enable_light_client_server`) I think we should return `ResourceUnavailable`.

Reply:
thank you, i am not familiar with lighthouse db layout